docs/design/architecture.md

```diff
@@ -2,128 +2,615 @@
+Kata Containers creates a VM in which to run one or more containers. It
+does this by launching a [hypervisor](#hypervisor) to create the VM.
+The hypervisor needs two assets for this task: a Linux kernel and a
+small root filesystem image to boot the VM.
 
 ### Guest kernel
 
-The guest kernel is passed to the hypervisor and used to boot the virtual
-machine. The default kernel provided in Kata Containers is highly optimized for
-kernel boot time and minimal memory footprint, providing only those services
-required by a container workload. This is based on a very current upstream Linux
-kernel.
+The [guest kernel](../../tools/packaging/kernel)
+is passed to the hypervisor and used to boot the VM.
+The default kernel provided in Kata Containers is highly optimized for
+kernel boot time and minimal memory footprint, providing only those
+services required by a container workload. It is based on the latest
+Linux LTS (Long Term Support) [kernel](https://www.kernel.org).
 
 ### Guest image
 
-Kata Containers supports both an `initrd` and `rootfs` based minimal guest image.
+The hypervisor uses an image file which provides a minimal root
+filesystem used by the guest kernel to boot the VM and host the Kata
+Container. Kata Containers supports both initrd and rootfs based
+minimal guest images. The [default packages](../install/) provide both
+an image and an initrd, both of which are created using the
+[`osbuilder`](../../tools/osbuilder) tool.
+
+> **Notes:**
+>
+> - Although initrd and rootfs based images are supported, not all
+>   [hypervisors](#hypervisor) support both types of image.
+>
+> - The guest image is *unrelated* to the image used in a container
+>   workload.
+>
+>   For example, if a user creates a container that runs a shell in a
+>   BusyBox image, they will run that shell in a BusyBox environment.
+>   However, the guest image running inside the VM that is used to
+>   *host* that BusyBox image could be running Clear Linux, Ubuntu,
+>   Fedora or any other distribution potentially.
+>
+>   The `osbuilder` tool provides
+>   [configurations for various common Linux distributions](../../tools/osbuilder/rootfs-builder)
+>   which can be built into either initrd or rootfs guest images.
+>
+> - If you are using a [packaged version of Kata
+>   Containers](../install), you can see image details by running the
+>   [`kata-collect-data.sh`](../../src/runtime/data/kata-collect-data.sh.in)
+>   script as `root` and looking at the "Image details" section of the
+>   output.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
+The default packaged rootfs image, sometimes referred to as the _mini
+O/S_, is a highly optimized container bootstrap system. On Intel
+x86_64 platforms, it is based on [Clear Linux](https://clearlinux.org)
```

Comment:
As I look at this again Clear Linux is recommended for rootfs but Alpine is recommended for initrd. Nothing really explains why not use the same for both.  I think adding a little clarification that Clear Linux is highly optimized and supports systemd init would help. I would clarify on the Alpine image below that it doesn't support systemd init which is why agent as the init works in that scenario. I thought on this repo somewhere there was a guide stating which OS images work for systemd rootfs and which ones work with agent as the init but I can't find it. If that guide does exist I would add a link to it here somewhere.